floor/floor/driver/raspberry.py

```diff
@@ -58,12 +60,58 @@ def __init__(self, args):
         self.spi.open(0, 0)
 
         self.weights = [0] * self.NUM_TILES
+        self.raw_weights = [0] * self.NUM_TILES
 
         # The maximum value seen after filtering.
         self.value_ceiling = [0] * self.NUM_TILES
         self.value_floor = [0] * self.NUM_TILES
 
         self.reader = SerialRead()
+        self.probe_floor()
+
+    def probe_floor(self):
```

Comment:
This can detect the floor configuration and is what I was working toward.  With this change it can tell if there are only 8 tiles connected.  A further change will to have it automatically pick an appropriate layout for the number of tiles.